lib/widgets/bottom_navigation.dart

```diff
@@ -14,35 +15,26 @@ class CocktailBottomNavigation extends ConsumerStatefulWidget {
 
 class _CocktailBottomNavigationState
     extends ConsumerState<CocktailBottomNavigation> {
-  int _selectedIndex = 0;
-
   @override
   Widget build(BuildContext context) {
+    int selectedIndex =
+        ref.read(homeViewModelProvider.notifier).currentNavigationIndex;
+
     return BottomNavigationBar(
       items: CocktailBottomNavigationType.values
           .map((type) => _buildBottomNavigationItem(type: type))
           .toList(),
-      currentIndex: _selectedIndex,
-      selectedItemColor: Colors.purple.shade300,
-      backgroundColor: Colors.amber.shade100,
-      onTap: (index) => onNavigationItemTap(index),
+      currentIndex: selectedIndex,
+      onTap: (index) {
+        ref.read(homeViewModelProvider.notifier).currentNavigationIndex = index;
+        context.router.push(CocktailBottomNavigationType.values[index].path);
```

Comment:
let's add this in a private function `_onTap`

Reply:
Done.

![image](https://user-images.githubusercontent.com/17646275/176975845-06aec508-62e8-414a-bf05-4a08151c1415.png)